echo-telemetry/src/main/java/com/netflix/spinnaker/echo/config/TelemetryConfig.kt

```
@@ -0,0 +1,87 @@
+import com.squareup.okhttp.OkHttpClient
+import de.huxhorn.sulky.ulid.ULID
+import java.util.concurrent.TimeUnit
+import org.slf4j.LoggerFactory
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import retrofit.RestAdapter
+import retrofit.client.OkClient
+import retrofit.converter.JacksonConverter
+
+@Configuration
+@ConditionalOnProperty("stats.enabled")
+@EnableConfigurationProperties(TelemetryConfigProps::class)
+open class TelemetryConfig {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(TelemetryConfig::class.java)
+  }
+
+  @Bean
+  open fun telemetryService(
+    retrofitConfigurationProperties: RetrofitConfigurationProperties,
+    configProps: TelemetryConfigProps
+  ): TelemetryService {
+    log.info("Telemetry service loaded")
+    return RestAdapter.Builder()
+      .setEndpoint(configProps.endpoint)
+      .setConverter(JacksonConverter())
+      .setClient(telemetryOkClient(configProps))
+      .setLogLevel(retrofitConfigurationProperties.logLevel)
+      .setLog(Slf4jRetrofitLogger(TelemetryService::class.java))
+      .build()
+      .create(TelemetryService::class.java)
+  }
+
+  private fun telemetryOkClient(configProps: TelemetryConfigProps): OkClient {
+    val httpClient = OkHttpClient()
+    httpClient.setConnectTimeout(configProps.connectionTimeoutMillis.toLong(), TimeUnit.MILLISECONDS)
+    httpClient.setReadTimeout(configProps.readTimeoutMillis.toLong(), TimeUnit.MILLISECONDS)
+    return OkClient(httpClient)
+  }
+
+  @ConfigurationProperties(prefix = "stats")
+  class TelemetryConfigProps {
+
+    companion object {
+      const val DEFAULT_TELEMETRY_ENDPOINT = "https://stats.spinnaker.io/log"
```

Should this be `https://stats.spinnaker.io/log` or `https://stats.spinnaker.io`?  Halyard sets it to `https://stats.spinnaker.io` without the `/log`...presumably we want to include the `/log`?  I guess that's not really related to these changes, but just checking.

---

Good catch! It looks like `/log` is appended to this URL by retrofit because we've set that as the path in [TelemetryService](https://github.com/spinnaker/echo/blob/3081a7f435b48dd75d970631bd99b112dde20ad4/echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/TelemetryService.java#L25). 

Confusingly, you get a `200` response no matter what `stats.spinnaker.io` URL you send data to. So I don't know if data sent to other paths gets recorded or silently discarded... but clearly `/log` is the intended URL, so I fixed it here, thanks!